rules/opentitan.bzl

```diff
@@ -112,7 +124,7 @@ sign_image = rule(
     },
 )
```

Comment:
Why is the platform `@local_config_platform//:host`?  I'd think this should be `OPENTITAN_PLATFORM` or that this rule wouldn't care about the toolchain transition at all (e.g. eliminate `cfg`, `platform` and `_allowlist_function_transition`).

The reason this rule might not care is that a downstream rule (like `elf_to_scrambled...`) causes the transition.

Reply:
The issue I have is:

1. the (existing) ROM_EXT signer tool should be built for the host platform, and is built with the [`rust_binary` rule](https://github.com/lowRISC/opentitan/blob/071283e7b1ed675807077751e689c94ed3707b87/sw/host/rom_ext_image_tools/signer/BUILD#L10), and

2. the ROM_EXT signer tool is a dependency of the `sign` bin rule, which takes as input targets built with the riscv toolchain (i.e., that transitioned to `OPENTITAN_PLATFORM`).

My (albeit very limited understanding) was that since the `sign_bin` rule sits downstream from rules that cause the transition to `OPENTITAN_PLATFORM`, that when the `sign_bin` rule goes to build the ROM_EXT signer tool, it needs to transition back to the host toolchain to do so.

Indeed, if I remove those lines as you suggest, I get an error:

<img width="1078" alt="Screen Shot 2022-03-02 at 1 42 02 PM" src="https://user-images.githubusercontent.com/5633066/156454372-cfadb5cb-8e55-42d0-9487-0cb69a3da592.png">

My solution was to transition to the host platform in the `sign_bin` rule so that the ROM_EXT signer could be built, then let further downstream rules transition back to the `OPENTITAN_PLATFORM`, since as you mention the `sign_bin` rule doesn't care about the toolchain per se, since its just running the signer tool on the bin file. (Note, the `elf_to_scrambed...` rule only applies to ROM images which are not signed. It is the `bin_to_flash_vmem` rule that takes as input signed bins.) This does indeed build successfully, but maybe not the most ideal way of doing it?

Ideally, it would be  nice if the transition could happen on the ROM_EXT signer tool  [`rust_binary` rule], but it didn't seem this was possible? Or is my understanding above completely wrong?

Reply:
(as discussed offline) I added a comment and issue (#11199) to track alternative solutions to this, so we can merge this in the meantime.